File: train.py
import tkinter as tk         # GUI用
import yolo11
import requests               # HTTPリクエスト用



# GUIウィンドウ作成
root = tk.Tk()
root.title("Hello Tkinter")
root.geometry("1200x1000")
image=tk.PhotoImage()  # 画像表示用
label = tk.Label(
    root,
    image=image,
    height=600,
    width=800,
)
label.pack()

def bbb():
    picture_requests = requests.get("http://localhost:3000/get_picture", params={"crossing-id":"test1"}, timeout=5)
    data = picture_requests.content
    image.config(data=data, format="png")
    word = yolo11.cls(data)
    return word
bbb()

if __name__ == '__main__':
    root.mainloop()

File: train.rb
require 'tk'
require 'net/http'
require 'uri'

# require 'serialport'
# https://rubydoc.info/gems/serialport/SerialPort

root = TkRoot.new { title "Train Camera"; geometry "400x300" }
images = TkPhotoImage.new
label = TkLabel.new(root) { image images; width 800; height 600}.pack

#sp = SerialPort.new('/dev/tty.usbmodem14201', 115200, 8, 1, 0)


def aaa
  puts "1秒経過"
  uri = URI("http://localhost:3000/vigcamera?crossing-id=test1")
  Net::HTTP.get(uri)
  Tk.after(1000) { bbb }
end

def bbb
  puts "1秒経過"
  uri = URI("http://localhost:3000/get_picture?crossing-id=test1")
  res = Net::HTTP.get(uri)
  File.binwrite(TEMP_FILE_NAME, res.body)
  images.read(TEMP_FILE_NAME, {"format": "png"})
  Tk.after(1000) { aaa }
end



# def put(data, *opts)


Tk.after(1000) { aaa }
Tk.mainloop

